src/test/java/seedu/sudohr/logic/commands/leave/DeleteEmployeeFromLeaveCommandTest.java

```diff
@@ -0,0 +1,413 @@
+package seedu.sudohr.logic.commands.leave;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static seedu.sudohr.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.sudohr.logic.commands.CommandTestUtil.VALID_LEAVE_DATE_LEAVE_TYPE_1;
+import static seedu.sudohr.testutil.Assert.assertThrows;
+
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.ObservableList;
+import javafx.collections.transformation.FilteredList;
+import seedu.sudohr.commons.core.GuiSettings;
+import seedu.sudohr.logic.commands.CommandResult;
+import seedu.sudohr.logic.commands.exceptions.CommandException;
+import seedu.sudohr.model.Model;
+import seedu.sudohr.model.ReadOnlySudoHr;
+import seedu.sudohr.model.ReadOnlyUserPrefs;
+import seedu.sudohr.model.SudoHr;
+import seedu.sudohr.model.department.Department;
+import seedu.sudohr.model.department.DepartmentName;
+import seedu.sudohr.model.employee.Employee;
+import seedu.sudohr.model.employee.Id;
+import seedu.sudohr.model.leave.Leave;
+import seedu.sudohr.model.leave.LeaveDate;
+import seedu.sudohr.testutil.TypicalEmployees;
+import seedu.sudohr.testutil.TypicalLeave;
+
+/**
+ * Deletes a employee from a specific leave in sudohr book.
+ */
+public class DeleteEmployeeFromLeaveCommandTest {
+
+    // Handle removing an employee
+    @Test
+    public void execute_employeeAcceptedByLeave_deleteSuccessful() throws CommandException {
+
+        ModelStubAcceptingEmployeeAdded modelStub = new ModelStubAcceptingEmployeeAdded();
+        modelStub.addEmployee(TypicalEmployees.ALICE);
+
+        new AddEmployeeToLeaveCommand(TypicalEmployees.ALICE_ID,
+                new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub);
+
+        CommandResult commandResult = new DeleteEmployeeFromLeaveCommand(
+                TypicalEmployees.ALICE_ID,
+                new LeaveDate(LocalDate.parse(VALID_LEAVE_DATE_LEAVE_TYPE_1))).execute(modelStub);
+
+        assertEquals(String.format(DeleteEmployeeFromLeaveCommand.MESSAGE_SUCCESS,
+                TypicalEmployees.ALICE, TypicalLeave.LEAVE_TYPE_1),
+                commandResult.getFeedbackToUser());
+
+        assertFalse(modelStub.sudoHr.getLeave(new Leave(new LeaveDate(LocalDate.parse(
+                VALID_LEAVE_DATE_LEAVE_TYPE_1))))
+                .hasEmployee(TypicalEmployees.ALICE));
+    }
+
```

Comment:
Can consider a diff test, suppose a user try to delete an invalid leave (e.g invalid format), which exception error will be thrown first? invalid format or leave does not exist?

Reply:
There is no need since LeaveDate cannot be formed if invalid format is used.  Case where leave does not exist is also tested